redis-utils-lib/src/cmr/redis_utils/redis.clj

```diff
@@ -6,39 +6,52 @@
    [cmr.redis-utils.config :as config]
    [taoensso.carmine :as carmine]))
 
-(defmacro wcar*
-  "Safe call redis with conn opts and retries.
-  For available commands refer to Redis docuemntation or:
-  https://github.com/ptaoussanis/carmine/blob/master/src/taoensso/carmine/commands.edn."
-  [& body]
-  `(let [with-retry#
-         (fn with-retry# [num-retries#]
-             (try
-               (carmine/wcar (config/redis-conn-opts) ~@body)
-               (catch Exception e#
-                 (if (> num-retries# 0)
-                   (do
-                     (info (format "Redis failed with exception %s. Retrying %d more times."
-                                   e#
-                                   (dec num-retries#)))
-                     (Thread/sleep (config/redis-retry-interval))
-                     (with-retry# (dec num-retries#)))
-                   (error "Redis failed with exception " e#)))))]
-     (with-retry# (config/redis-num-retries))))
+  (defmacro wcar*
+    "Safe call redis with conn opts and retries.
+    The passed in key allows for specific redis caches as well as more logging information.
+    The read? is a true/false flag that signifies that a read-only replica redis connection
+    should be used. This flag is also used in more logging information.
+    For available commands refer to Redis docuemntation or:
+    https://github.com/ptaoussanis/carmine/blob/master/src/taoensso/carmine/commands.edn."
+    [key read? & body]
+    `(let [with-retry#
+            (fn with-retry# [num-retries#]
+              (let [wr-string# (if ~read? "read" "write")
+                    config# (if (= :collection-metadata-cache ~key)
```

Comment:
The only usages of wcar* I saw outside of the defrecord RedisCache and defrecord RedisHashCache was for tests (as wcar* and the get-keys function in the same file) and in dev-system/src/cmr/dev_system/control.clj for the /reset and /clear-cache endpoints (with the reset function from this file).
Would we be able to add the connection options as a part of the redis cache record details and then just pass the connection options into wcar* as an argument so we can avoid this logic?